### ISSUE TITLE
Comprehensive Benchmark: AMP Prediction Tools Comparison

### ISSUE THREAD

**Issue by madsondeluna** [madsondeluna/AMPidentifier#5]
<html><head></head><body><h1>Comprehensive Benchmark: AMP Prediction Tools Comparison</h1>
<h2>Summary</h2>
<p>This document provides a detailed comparative analysis of antimicrobial peptide (AMP) prediction tools, including AMPidentifier's performance metrics against state-of-the-art methods. This benchmark is intended for publication and documentation purposes.</p>
<h2>Motivation</h2>
<p>To establish AMPidentifier's position within the current landscape of AMP prediction tools, this comprehensive comparison includes:</p>
<ul>
<li>18 AMP prediction tools (11 primary + 7 additional)</li>
<li>Detailed technical specifications</li>
<li>FAIR compliance assessment</li>
<li>Performance metrics from standardized benchmarks</li>
<li>Implementation characteristics</li>
</ul>
<hr>
<h2>AMPidentifier Performance Summary</h2>
<p><strong>Tested on balanced datasets:</strong></p>

Activity Type | Dataset Size | Accuracy | Sensitivity | Specificity | TP | FN | TN | FP
-- | -- | -- | -- | -- | -- | -- | -- | --
Antibacterial | 1,968 (984 pos, 984 neg) | 82.16% | 88.41% | 75.91% | 870 | 114 | 747 | 237
Antifungal | 2,768 (1,384 pos, 1,384 neg) | 80.81% | 86.35% | 75.29% | 1,195 | 189 | 1,042 | 342
Antiviral | 1,478 (739 pos, 739 neg) | 61.37% | 45.06% | 77.67% | 333 | 406 | 574 | 165
Antimicrobial (antibac+antifung) | - | 81.38% | 87.21% | 75.55% | - | - | - | -

<hr>
<h2>FAIR Compliance Criteria</h2>
<p><strong>Classification System:</strong></p>
<ul>
<li><strong>Yes (Fully FAIR-compliant)</strong>: Open-source code in public repository (GitHub/CRAN/Bioconda), clear license (MIT/GPL/Apache), complete documentation, standard formats (FASTA), peer-reviewed publication with DOI</li>
<li><strong>Partial</strong>: Code available but limited documentation, OR web-only without CLI/API, OR partial code release</li>
<li><strong>No</strong>: Web-only without public code, no clear license, inadequate documentation/API</li>
</ul>
<p><strong>Fully FAIR-compliant tools (6/11 primary):</strong></p>
<ol>
<li>AMPidentifier</li>
<li>AMPlify</li>
<li>amPEPpy</li>
<li>Macrel</li>
<li>ampir</li>
<li>AmpGram</li>
</ol>
<hr>
<h2>Feature Descriptors Glossary</h2>
<p>Common feature types used across tools:</p>
<ul>
<li><strong>AAC</strong>: Amino Acid Composition</li>
<li><strong>DPC</strong>: Dipeptide Composition</li>
<li><strong>PAAC</strong>: Pseudo Amino Acid Composition</li>
<li><strong>PseAAC</strong>: Pseudo-Amino Acid Composition</li>
<li><strong>NAAC</strong>: Normalized Amino Acid Composition</li>
<li><strong>PHYC</strong>: Physicochemical properties</li>
<li><strong>STRL</strong>: Structural features</li>
<li><strong>PseKRAAC</strong>: Pseudo K-tuple Reduced Amino Acid Composition</li>
<li><strong>FET</strong>: Free Energy Transition</li>
<li><strong>PC6</strong>: 6 physicochemical properties encoding</li>
</ul>
<hr>
<h2>Training Dataset Characteristics</h2>
<p><strong>Important Considerations:</strong></p>
<ol>
<li><strong>Dataset sizes vary significantly</strong> across tools (from ~800 to 102,756+ sequences)</li>
<li><strong>Common benchmarks</strong>:
<ul>
<li>Xiao et al. 2013 (920 AMPs + 920 non-AMPs) - most widely used</li>
<li>APD3 database</li>
</ul>
</li>
<li><strong>Balanced vs imbalanced datasets</strong> affect real-world performance</li>
<li><strong>Larger training sets</strong> (&gt;3,000 sequences) generally produce more robust models</li>
<li><strong>Activity-specific datasets</strong> (antibacterial, antifungal, antiviral) enable specialized predictions</li>
</ol>
<hr>
<h2>Performance Analysis</h2>
<h3>Accuracy Metrics</h3>
<p><strong>High performers (&gt;90% on benchmarks):</strong></p>
<ul>
<li>amPEPpy: 96.0%</li>
<li>AMPScanner v2: ~95.5%</li>
<li>AMPlify: 93.71%</li>
<li>AI4AMP: 90.0%</li>
</ul>
<p><strong>Good performers (80-90%):</strong></p>
<ul>
<li>AMPidentifier: 82.16% (antibacterial), 80.81% (antifungal)</li>
<li>iAMPpred: ~85.0%</li>
<li>APIN: ~85.0%</li>
<li>CS-AMPPred: ~82.0%</li>
<li>iAMP-2L: ~80.0%</li>
<li>MLAMP: ~80.0%</li>
</ul>
<p><strong>Specialized performers:</strong></p>
<ul>
<li>Deep-AmPEP30: 77.0% (optimized for short peptides ≤30aa)</li>
</ul>
<h3>Key Performance Notes</h3>
<ol>
<li><strong>Reported accuracy varies with test dataset</strong> used</li>
<li><strong>Xiao et al. 2013 benchmark</strong> used by majority for comparison</li>
<li><strong>AMPidentifier tested on activity-specific datasets</strong> (antibacterial, antifungal, antiviral) - unique approach</li>
<li><strong>Recent tools (2020+)</strong> tend to use deep learning with improved performance</li>
<li><strong>Ensemble methods</strong> typically outperform single models</li>
</ol>
<hr>
<h2>Evolutionary Trends in AMP Prediction</h2>
<h3>Generation 1 (2012-2016): Classical ML</h3>
<ul>
<li>Traditional machine learning (SVM, RF) with manual feature engineering</li>
<li>Focus on composition and physicochemical features</li>
<li>Tools: ClassAMP, iAMP-2L, CS-AMPPred, CAMPR3</li>
</ul>
<h3>Generation 2 (2017-2019): Feature Enhancement</h3>
<ul>
<li>Improved feature sets and ensemble methods</li>
<li>Introduction of pseudo-amino acid compositions</li>
<li>Tools: iAMPpred, amPEPpy, APIN, PEPred-Suite</li>
</ul>
<h3>Generation 3 (2020+): Deep Learning Era</h3>
<ul>
<li>Deep learning architectures (CNN, RNN, LSTM, Attention mechanisms)</li>
<li>Sequence embeddings and learned representations</li>
<li>End-to-end learning without manual feature engineering</li>
<li>Tools: Macrel, ampir, AmpGram, AI4AMP, AMPlify</li>
</ul>
<h3>Generation 4 (2022+): Specialized &amp; Ensemble</h3>
<ul>
<li>Activity-specific models (antibacterial, antifungal, antiviral)</li>
<li>Ensemble of diverse architectures</li>
<li>Focus on FAIR principles and reproducibility</li>
<li>Tools: AMPlify, AMPidentifier</li>
</ul>
<hr>
<h2>Open Source &amp; FAIR Trends</h2>
<p><strong>Positive developments:</strong></p>
<ul>
<li>6/11 primary tools are fully FAIR-compliant (55%)</li>
<li>Growing trend of depositing code on GitHub</li>
<li>CLI/Pip/Conda tools show higher adoption rates</li>
<li>Increased focus on reproducibility and transparency</li>
</ul>
<p><strong>Challenges:</strong></p>
<ul>
<li>Several high-performing tools remain proprietary (AI4AMP, iAMP-2L)</li>
<li>Web-only tools limit integration into pipelines</li>
<li>Inconsistent documentation quality</li>
<li>Variable maintenance and update frequency</li>
</ul>
<hr>
<h2>Tool Specialization Characteristics</h2>
<p><strong>By prediction target:</strong></p>
<ul>
<li><strong>Mature peptides</strong>: ampir (mature model), AMPlify</li>
<li><strong>Precursor proteins</strong>: ampir (precursor model)</li>
<li><strong>Short peptides (≤30aa)</strong>: Deep-AmPEP30</li>
<li><strong>Cysteine-rich AMPs</strong>: CS-AMPPred</li>
<li><strong>Longer AMPs/proteins</strong>: AmpGram</li>
<li><strong>Activity-specific</strong>: AMPidentifier (antibacterial, antifungal, antiviral)</li>
</ul>
<p><strong>By application:</strong></p>
<ul>
<li><strong>Genome-wide screening</strong>: Macrel, ampir (precursor)</li>
<li><strong>High-throughput prediction</strong>: AMPlify, Macrel</li>
<li><strong>Database integration</strong>: DBAASP, CAMPR3</li>
<li><strong>Multi-activity prediction</strong>: MLAMP, PEPred-Suite</li>
<li><strong>Hemolytic activity</strong>: Macrel</li>
</ul>
<hr>
<h2>Usage Recommendations</h2>
<h3>For Different Use Cases</h3>
<p><strong>High-throughput genome screening:</strong></p>
<ul>
<li>Primary: Macrel, ampir (precursor model)</li>
<li>Alternative: AMPlify (imbalanced model)</li>
</ul>
<p><strong>Mature peptide prediction:</strong></p>
<ul>
<li>Primary: ampir (mature model), AMPlify</li>
<li>Alternative: amPEPpy, AMPidentifier</li>
</ul>
<p><strong>Activity-specific prediction:</strong></p>
<ul>
<li>Antibacterial: AMPidentifier</li>
<li>Antifungal: AMPidentifier</li>
<li>Antiviral: AMPlify, AI4AMP (AMPidentifier shows limitations)</li>
</ul>
<p><strong>Cysteine-rich AMPs:</strong></p>
<ul>
<li>Primary: CS-AMPPred</li>
</ul>
<p><strong>Research &amp; development:</strong></p>
<ul>
<li>AMPlify (best documented, FAIR-compliant)</li>
<li>AMPidentifier (activity-specific, ensemble)</li>
<li>Macrel (genome/metagenome focus)</li>
</ul>
<p><strong>Quick web-based prediction:</strong></p>
<ul>
<li>AMPScanner v2 (high accuracy)</li>
<li>AI4AMP (good performance)</li>
<li>CAMPR3 (multiple algorithms)</li>
</ul>
<h3>Integration Considerations</h3>
<p><strong>For pipeline integration:</strong></p>
<ol>
<li>CLI/Pip tools: AMPidentifier, AMPlify, amPEPpy, Macrel</li>
<li>R/CRAN: ampir, AmpGram</li>
<li>Docker/Conda: Macrel, AMPlify</li>
<li>API access: AMPidentifier, AMPlify, Macrel</li>
</ol>
<p><strong>For reproducibility:</strong></p>
<ul>
<li>Prefer FAIR-compliant tools with public repositories</li>
<li>Document exact versions and parameters used</li>
<li>Use standardized benchmarks for comparison</li>
<li>Report both balanced and imbalanced test results</li>
</ul>
<hr>
<h2>AMPidentifier's Unique Contributions</h2>
<ol>
<li><strong>Activity-specific models</strong>: Separate predictions for antibacterial, antifungal, and antiviral activities</li>
<li><strong>Ensemble architecture</strong>: Combines RF, SVM, and GB for robust predictions</li>
<li><strong>FAIR compliance</strong>: Fully open-source with MIT license</li>
<li><strong>Registered software</strong>: INPI registration (BR512025005859-4)</li>
<li><strong>Multiple interfaces</strong>: CLI, Pip package, and web server</li>
<li><strong>Balanced datasets</strong>: Carefully curated balanced test sets per activity type</li>
</ol>
<p><strong>Areas for improvement</strong> (identified from this benchmark):</p>
<ul>
<li>Antiviral prediction sensitivity (45.06%) needs enhancement</li>
<li>Consider adding hemolytic activity prediction</li>
<li>Explore deep learning architectures for comparison</li>
<li>Expand training dataset size (currently 1-3K per activity)</li>
</ul>
<hr>
<h2>Implementation Roadmap</h2>
<h3>Short-term enhancements</h3>
<ul>
<li>[ ] Add this benchmark table to README.md</li>
<li>[ ] Create comparison plots (ROC curves, bar charts)</li>
<li>[ ] Document testing methodology in detail</li>
<li>[ ] Publish benchmark datasets for reproducibility</li>
<li>[ ] Add performance comparison script</li>
</ul>
<h3>Medium-term improvements</h3>
<ul>
<li>[ ] Improve antiviral prediction (consider transfer learning from AMPlify/AI4AMP)</li>
<li>[ ] Add hemolytic activity prediction module</li>
<li>[ ] Implement deep learning baseline for comparison</li>
<li>[ ] Expand training datasets (target 5K+ per activity)</li>
<li>[ ] Add batch processing optimizations</li>
</ul>
<h3>Long-term goals</h3>
<ul>
<li>[ ] Multi-species specific models</li>
<li>[ ] Minimum inhibitory concentration (MIC) prediction</li>
<li>[ ] Structure-based predictions (integrate AlphaFold)</li>
<li>[ ] Active learning module for continuous improvement</li>
<li>[ ] Community benchmark challenges</li>
</ul>
<hr>
<h2>References</h2>
<h3>Primary Tool Publications</h3>
<ol>
<li>
<p>Veltri D, Kamath U, Shehu A. Deep learning improves antimicrobial peptide recognition. <em>Bioinformatics</em>. 2018;34(16):2740-2747. doi:10.1093/bioinformatics/bty179</p>
</li>
<li>
<p>Li C, Sutherland D, Hammond SA, et al. AMPlify: attentive deep learning model for discovery of novel antimicrobial peptides effective against WHO priority pathogens. <em>BMC Genomics</em>. 2022;23:77. doi:10.1186/s12864-022-08310-4</p>
</li>
<li>
<p>Santos-Júnior CD, Pan S, Zhao X, Coelho LP. Macrel: antimicrobial peptide screening in genomes and metagenomes. <em>PeerJ</em>. 2020;8:e10555. doi:10.7717/peerj.10555</p>
</li>
<li>
<p>Fingerhut LCHW, Miller DJ, Strugnell JM, Daly NL, Cooke IR. ampir: an R package for fast genome-wide prediction of antimicrobial peptides. <em>Bioinformatics</em>. 2020;36(21):5262-5263. doi:10.1093/bioinformatics/btaa653</p>
</li>
<li>
<p>Meher PK, Sahu TK, Saini V, Rao AR. Predicting antimicrobial peptides with improved accuracy by incorporating the compositional, physico-chemical and structural features into Chou's general PseAAC. <em>Sci Rep</em>. 2017;7:42362. doi:10.1038/srep42362</p>
</li>
<li>
<p>Xiao X, Wang P, Lin WZ, Jia JH, Chou KC. iAMP-2L: a two-level multi-label classifier for identifying antimicrobial peptides and their functional types. <em>Anal Biochem</em>. 2013;436(2):168-177. doi:10.1016/j.ab.2013.01.019</p>
</li>
<li>
<p>Bhadra P, Yan J, Li J, et al. AmPEP: Sequence-based prediction of antimicrobial peptides using distribution patterns of amino acid properties and random forest. <em>Sci Rep</em>. 2018;8:1697. doi:10.1038/s41598-018-19752-w</p>
</li>
<li>
<p>Yan J, Bhadra P, Li A, et al. Deep-AmPEP30: Improve Short Antimicrobial Peptides Prediction with Deep Learning. <em>Mol Ther Nucleic Acids</em>. 2020;20:882-894. doi:10.1016/j.omtn.2020.05.006</p>
</li>
<li>
<p>Burdukiewicz M, Sidorczuk K, Rafacz D, et al. Proteomic Screening for Prediction and Design of Antimicrobial Peptides with AmpGram. <em>Int J Mol Sci</em>. 2020;21(12):4310. doi:10.3390/ijms21124310</p>
</li>
<li>
<p>Fingerhut LCHW, Miller DJ, Strugnell JM, Daly NL, Cooke IR. ampir: an R package for fast genome-wide prediction of antimicrobial peptides. <em>Bioinformatics</em>. 2020;36(21):5262-5263. doi:10.1093/bioinformatics/btaa653</p>
</li>
</ol>
<h3>Benchmark Datasets</h3>
<ol start="11">
<li>
<p>Xiao X, Wang P, Lin WZ, Jia JH, Chou KC. iAMP-2L: A two-level multi-label classifier for identifying antimicrobial peptides and their functional types. <em>Anal Biochem</em>. 2013;436(2):168-177.</p>
</li>
<li>
<p>Wang G, Li X, Wang Z. APD3: the antimicrobial peptide database as a tool for research and education. <em>Nucleic Acids Res</em>. 2016;44(D1):D1087-D1093. doi:10.1093/nar/gkv1278</p>
</li>
</ol>
<h3>Reviews &amp; Comparisons</h3>
<ol start="13">
<li>
<p>Gabere MN, Noble WS. Empirical comparison of web-based antimicrobial peptide prediction tools. <em>Bioinformatics</em>. 2017;33(13):1921-1929. doi:10.1093/bioinformatics/btx081</p>
</li>
<li>
<p>Jhong JH, Chi YH, Li WC, et al. dbAMP: an integrated resource for exploring antimicrobial peptides with functional activities and physicochemical properties on transcriptome and proteome data. <em>Nucleic Acids Res</em>. 2019;47(D1):D285-D297. doi:10.1093/nar/gky1030</p>
</li>
</ol>
<hr>
<h2>Data Availability</h2>
<p>All benchmark data, scripts, and supplementary materials will be made available at:</p>
<ul>
<li>GitHub repository: https://github.com/madsondeluna/ampidentifier</li>
<li>Zenodo archive: [To be created upon publication]</li>
</ul>
<hr>
<h2>Acknowledgments</h2>
<p>This comprehensive benchmark was compiled from publicly available literature, tool documentation, and official repositories. We acknowledge all tool developers for their contributions to the AMP prediction field.</p>
<hr>
<h2>License</h2>
<p>This benchmark document is released under CC-BY 4.0 license.</p>
<hr>
<p><strong>Last Updated</strong>: 2025-03-18<br>
<strong>Version</strong>: 1.0<br>
<strong>Maintainer</strong>: Madson A. de Luna Aragão (@madsondeluna)</p></body></html>
